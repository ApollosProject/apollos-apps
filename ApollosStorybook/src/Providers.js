import React from 'react';
import ApollosConfig from '@apollosproject/config';
import { Providers } from '@apollosproject/ui-kit';
import { MediaPlayerProvider } from '@apollosproject/ui-media-player';
import { LiveProvider } from '@apollosproject/ui-connected';
import ClientProvider, { client } from './client';
import customTheme, { customIcons } from './theme';

const AppProviders = (props) => (
  <ClientProvider>
    <MediaPlayerProvider>
        <LiveProvider>
          <Providers
            themeInput={customTheme}
            iconInput={customIcons}
            {...props}
          />
        </LiveProvider>
    </MediaPlayerProvider>
  </ClientProvider>
);

export default AppProviders;
