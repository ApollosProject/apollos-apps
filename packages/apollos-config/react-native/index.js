import NativeConfig from 'react-native-config';

class Config {
  config = {
    ...NativeConfig,
  };

  constructor() {
    this.attachConfigToClass(this.config);
  }

  loadJs(data = {}) {
    Object.keys(data).forEach((key) => {
      if (this.config[key] && typeof this.config[key] === 'object') {
        // Deep merge defaults if they exist.
        this.config[key] = Object.assign(this.config[key], data[key]);
      } else {
        // Otherwise attach results to config.
        this.config[key] = data[key];
      }
    });
    this.attachConfigToClass(this.config);
  }

  // Allows you to do Config.PROPERTY
  attachConfigToClass(config) {
    Object.keys(config).forEach((key) => {
      Object.defineProperty(this, key, {
        enumerable: true,
        configurable: true,
        writable: true,
        value: config[key],
      });
    });
  }
}

const config = new Config();

export default config;
