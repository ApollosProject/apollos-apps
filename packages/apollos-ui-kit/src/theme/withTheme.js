import PropTypes from 'prop-types';
import { compose, mapProps, getContext, hoistStatics } from 'recompose';
import { get } from 'lodash';

import mergeStyles from '../styled/mergeStyles';
import { THEME_PROPS } from './createTheme';

const DEFAULT_MAPPER_FN = ({ theme } = {}) => ({ theme });

export default function(mapperFn = DEFAULT_MAPPER_FN, fqn) {
  return hoistStatics(
    compose(
      getContext({
        theme: PropTypes.shape(THEME_PROPS),
      }),
      mapProps(({ theme, ...otherProps }) => {
        const themeOverridesValue = fqn
          ? get(theme, `overrides['${fqn}']`, {})
          : {};

        const themeOverrides =
          typeof themeOverridesValue === 'function'
            ? themeOverridesValue(otherProps)
            : themeOverridesValue;

        const withThemeProps = mapperFn({ theme, ...otherProps });

        let mergedOverides = {};
        // We need to deep merge te style props. Other nested props are not currently merged
        if (themeOverrides.style && withThemeProps.style) {
          const mergedStyles = mergeStyles(
            withThemeProps.style,
            themeOverrides.style
          );
          mergedOverides = { style: mergedStyles };
        }

        return {
          ...otherProps,
          ...withThemeProps,
          ...themeOverrides,
          ...mergedOverides,
        };
      })
    )
  );
}
