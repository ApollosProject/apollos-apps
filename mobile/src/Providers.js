import { nest } from 'recompose';

import ClientProvider from '/mobile/client';
import { ThemeProvider } from '/mobile/ui/theme';
import { WebBrowserProvider } from '/mobile/ui/WebBrowser';

export default nest(ClientProvider, ThemeProvider, WebBrowserProvider);
