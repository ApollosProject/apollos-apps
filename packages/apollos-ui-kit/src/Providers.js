import { nest } from 'recompose';

import { ThemeProvider } from './theme';
import { LayoutProvider } from './LayoutContext';

export default nest(ThemeProvider, LayoutProvider);
