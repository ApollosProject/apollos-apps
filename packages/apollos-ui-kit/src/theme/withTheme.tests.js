import React from 'react';
import renderer from 'react-test-renderer';
import { View } from 'react-native';

import Providers from '../Providers';

import withTheme from './withTheme';

describe('the styled HOC', () => {
  it('provides a theme', () => {
    const StyledView = withTheme(({ theme }) => ({
      style: {
        backgroundColor: theme.colors.primary,
      },
      clickable: false,
    }))(View);
    const tree = renderer.create(
      <Providers>
        <StyledView />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('allows a theme to supply overrides', () => {
    const StyledView = withTheme(
      ({ theme }) => ({
        style: {
          backgroundColor: theme.colors.primary,
        },
        clickable: false,
      }),
      'StyledView'
    )(View);
    const overrides = {
      StyledView: { style: { backgroundColor: 'green' }, clickable: true },
    };
    const tree = renderer.create(
      <Providers themeInput={{ overrides }}>
        <StyledView />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('allows a theme to supply override functions', () => {
    const StyledView = withTheme(
      ({ theme }) => ({
        style: {
          backgroundColor: theme.colors.primary,
        },
        clickable: false,
      }),
      'StyledView'
    )(View);
    const overrides = {
      StyledView: () => ({ active }) => ({
        style: { backgroundColor: active ? 'active-color' : 'in-active-color' },
        clickable: active,
      }),
    };
    const tree = renderer.create(
      <Providers themeInput={{ overrides }}>
        <StyledView active />
        <StyledView />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('merges nested styles from withTheme and an overide', () => {
    const StyledView = withTheme(
      ({ theme }) => ({
        style: {
          backgroundColor: theme.colors.primary,
          color: 'blue',
        },
        clickable: false,
      }),
      'StyledView'
    )(View);
    const overrides = {
      StyledView: {
        style: {
          backgroundColor: 'active-color',
          borderRadius: 7,
        },
        clickable: false,
      },
    };
    const tree = renderer.create(
      <Providers themeInput={{ overrides }}>
        <StyledView />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
