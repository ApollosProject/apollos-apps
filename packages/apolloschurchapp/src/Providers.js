import { nest } from 'recompose';

import ClientProvider from 'apolloschurchapp/src/client';
import { ThemeProvider } from 'apolloschurchapp/src/ui/theme';
import { WebBrowserProvider } from 'apolloschurchapp/src/ui/WebBrowser';

export default nest(ClientProvider, ThemeProvider, WebBrowserProvider);
