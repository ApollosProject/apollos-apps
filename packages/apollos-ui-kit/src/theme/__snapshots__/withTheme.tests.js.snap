// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the styled HOC allows a theme to supply override functions 1`] = `
Array [
  <View
    active={true}
    clickable={true}
    style={
      Object {
        "backgroundColor": "active-color",
      }
    }
  />,
  <View
    style={
      Object {
        "backgroundColor": "in-active-color",
      }
    }
  />,
]
`;

exports[`the styled HOC allows a theme to supply overrides 1`] = `
<View
  clickable={true}
  style={
    Object {
      "backgroundColor": "green",
    }
  }
/>
`;

exports[`the styled HOC merges nested styles from withTheme and an overide 1`] = `
<View
  clickable={false}
  style={
    Object {
      "backgroundColor": "active-color",
      "borderRadius": 7,
      "color": "blue",
    }
  }
/>
`;

exports[`the styled HOC provides a theme 1`] = `
<View
  clickable={false}
  style={
    Object {
      "backgroundColor": "#00676D",
    }
  }
/>
`;
